multicore_runtime/dispatcher.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2015 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""A WSGI app that, once configured, dispatches requests to user apps."""
+
+import logging
+import re
+
+from werkzeug.wrappers import Request
+from werkzeug.wrappers import Response
+
+
+def dispatcher(handlers):
+  """Accepts handlers and returns a WSGI app that dispatches requests to them.
+
+  Args:
+    handlers: a list of handlers as produced by
+      wsgi_utils.load_user_scripts_into_handlers: a list of tuples of
+      (url, script, app).
+
+  Returns:
+    A WSGI app that dispatches to the user apps specified in the input.
+  """
+
+  def dispatch(wsgi_env, start_response):
+    """Handle one request."""
+    request = Request(wsgi_env)
+    for url, script, app in handlers:  # pylint: disable=unused-variable
+      matcher = re.match(url, request.path)
+      if matcher and matcher.end() == len(request.path):
+        if app is not None:
```

Comment:
Recommendation is to use "implicit" false, which tends to be more pythonic:
https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=True/False_evaluations#True/False_evaluations

In some cases I agree the explicit `is not None` is more appropriate, but in this case I think it's unlikely (impossible?) that `app` will hold a value other than `None` that evaluates to False in a boolean context, so `if app:` is probably appropriate here.